torch/csrc/utils/python_arg_parser.h

```diff
@@ -79,82 +79,6 @@
 #include <string>
 #include <vector>
 
-namespace pybind11 {
-namespace detail {
-template <>
-struct type_caster<c10::SymInt> {
- public:
-  PYBIND11_TYPE_CASTER(c10::SymInt, _("SymInt"));
```

Comment:
Wait the lint didn't catch that??

Reply:
PYBIND11_SPECIALIZATION lint only checks for PYBIND11_DECLARE_HOLDER_TYPE. It is kind of annoying to make it apply to type caster as well

Reply:
also, the old caster was in this file and didn't cause any problems. Very perplexing.